skimage/graph/tests/test_rag.py

```diff
@@ -108,3 +108,52 @@ def test_rag_error():
     labels[5:, :] = 1
     testing.assert_raises(ValueError, graph.rag_mean_color, img, labels,
                           2, 'non existant mode')
+
+
+def _weight_mean_color(graph, src, dst, n):
+    diff = graph.node[dst]['mean color'] - graph.node[n]['mean color']
+    diff = np.linalg.norm(diff)
+    return diff
+
+
+def _pre_merge_mean_color(graph, src, dst):
+    graph.node[dst]['total color'] += graph.node[src]['total color']
+    graph.node[dst]['pixel count'] += graph.node[src]['pixel count']
+    graph.node[dst]['mean color'] = (graph.node[dst]['total color'] /
+                                     graph.node[dst]['pixel count'])
+
+
+def merge_hierarchical_mean_color(labels, rag, thresh, rag_copy=True,
+                                  in_place_merge=False):
+    return graph.merge_hierarchical(labels, rag, thresh, rag_copy,
+                                    in_place_merge, _pre_merge_mean_color,
+                                    _weight_mean_color)
+
+
+@skipif(not is_installed('networkx'))
+def test_rag_hierarchical():
+    img = np.zeros((8, 8, 3), dtype='uint8')
+    labels = np.zeros((8, 8), dtype='uint8')
+
+    img[:, :, :] = 30
+    labels[:, :] = 1
+
+    img[0:4, 0:4, :] = 10, 10, 10
+    labels[0:4, 0:4] = 2
+
+    img[4:, 0:4, :] = 20, 20, 20
+    labels[4:, 0:4] = 3
+
```

Comment:
@vighneshbirodkar But I want you to make the test more stringent: `len(unique(labels))` can hide many things going wrong. I want you to check that `result` is equal to an `expected_result` array. So I wanted to make sure that 10 and 20 merge before 20 and 30, reliably and logically. =) One alternative to stay with uint8 is to use (11 * sqrt(3) + eps) as the threshold and then use 10, 20, 31.